### ISSUE TITLE
PRs from openapi-sdkautomation should not be marked as customer-reported

### ISSUE THREAD

**Issue by ramya-rao-a** [Azure/azure-sdk-tools#1354]
These are the PRs from automation that end up getting the `customer-reported` label. See https://github.com/Azure/azure-sdk-for-js/pulls?q=is%3Apr+author%3Aapp%2Fopenapi-sdkautomation+is%3Aclosed+

This messes up my count for actual community contributions :(

**Comment by ramya-rao-a** [comment]
cc @AlexGhiondea, @mitchdenny

**Comment by kurtzeborn** [comment]
Assigning this to @lillyjma for follow up.  Pretty sure the change is to FabricBot rules to not apply that label when the issues are opened by a bot.  Not sure if FabricBot is able to distinguish between a bot account and a real account (note that in GitHub UI it's labeled as a "bot" account).  If it can't, a workaround for this specific issue is to not apply the label when this specific bot account ("openapi-sdkautomation") opened the issue.

**Comment by lilyjma** [comment]
@AlexGhiondea: This bot is called `openapi-sdkautomation`, not the usual `msftbot` that I'm familiar with. Not sure who owns this bot or when it's added to our repos.

**Comment by mitchdenny** [comment]
The ```openapi-sdkautomation``` bot is something that creates PRs against our repository to validate incoming swagger changes. However the intent behind this issue is really to make sure _our_ fabric bot knows not to set those PRs/issues as customer reported.

We don't know for sure, but the rule you need to set is probably something like this:

![image](https://user-images.githubusercontent.com/513398/107300292-56adc200-6acd-11eb-9de9-1e84524de5b9.png)

You'll need to experiment to make sure.

**Comment by lilyjma** [comment]
@mitchdenny Yeah, I figured that might be the problem. Wes said the openapi bot isn't even a user account, so I'm not sure if this will help. He suspects that the bot is a GH app, and we're waiting for Phoenix to confirm that. I looked through all the conditions available, if the bot is an app, maybe we could do: Condition="Title contains..." Pattern="ReleasePR". This is not ideal, but I can't think of anything else now. @AlexGhiondea thoughts?

**Comment by ramya-rao-a** [comment]
I would not recommend using any pattern in the PR title which is subject to change at any point.

**Comment by mitchdenny** [comment]
I'm pretty sure the GitHub events that FabricBot is responding to will have the bot name where the user identity is.

**Comment by lilyjma** [comment]
@mitchdenny I didn't know that's the case. I've added the rule that you proposed. @ramya-rao-a let me know if the issue still comes up.

**Comment by PhoenixHe-msft** [comment]
@zhenglaizhang  @ruowan to confirm the config. I don't know much about this config line.

**Comment by ruowan** [comment]
We didn't own this config. Maybe it could work by adding a new check for activity sender.

**Comment by lilyjma** [comment]
@mitchdenny I don't think it's working. I added the rule 2 days ago, and for [a recent PR](https://github.com/Azure/azure-sdk-for-js/pull/13848) created by the api bot, the msftbot is still adding customer-report. The api bot is actually an app, https://github.com/apps/openapi-sdkautomation. Maybe that's why adding the condition didn't work.

**Comment by ramya-rao-a** [comment]
Any updates here?

**Comment by ramya-rao-a** [comment]
Ping!

**Comment by ramya-rao-a** [comment]
@AlexGhiondea mentioned offline that he has pushed something today that should fix this. I will remove the label for all the older PRs and then wait to see if the new one is getting the label or not

**Comment by weshaggard** [comment]
I'm not sure if this will help or not but I noticed in the github issues query it marked the user as "app/openapi-sdkautomation" so I tried using that as the user.
![image](https://user-images.githubusercontent.com/9010698/132067144-a30b1cf7-5ed1-449e-ae9f-25cb4b638bc1.png)

@ramya-rao-a please let use know if you see any new issues.

**Comment by ramya-rao-a** [comment]
@weshaggard This PR is from 4 days ago and it did get the customer-reported label: https://github.com/Azure/azure-sdk-for-js/pull/17566

**Comment by weshaggard** [comment]
I suspect we don't have many other options here given the tools we have. While looking at the rule again I did notice there was a space at the end of the user name, not sure if that is any issue or not but I removed the space and saved the rule just in case.

**Comment by AlexGhiondea** [comment]
@weshaggard that might have done it... let's wait and see. If that doesn't work, I am not sure what else we can do... maybe have a separate rule that removes that label?

**Comment by xiangyan99** [comment]
We are seeing the same issue in python repo. e.g. https://github.com/Azure/azure-sdk-for-python/pull/21251.

It was created by azclibot and we labeled it "customer-reported".

**Comment by weshaggard** [comment]
There are some bot accounts that are github users, but there are also github applications that have their own set of permissions and can create issues/PRs in the repo and they are app accounts.
